transcom/mymove
# INT B-18829 Admin UI show requested office user overview page

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18829)

## Summary

When requested office users are requesting accounts, admin users will need to be able to see an overview of that user before approving/rejecting/editing (similar to how office users are edited in the admin UI currently).

Since the admin app uses `react-admin`, we simply throw that endpoint in there and add it to the current Requested Office Users resource.

I did make some updates to the backend to allow for passing in the roles in the `GET` request.

Also you'll see there's an `EDIT` button on this page that currently shows the edit form for an approved office user. This will be changed in [B-18832](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18832) to show the relevant edit form for requested office users instead. The way `react-admin` works is that it only shows that edit button if there is an edit parameter in the resource.

### How to test

1. Access MM as an admin
2. Login and you should see a list of requested users
3. If you don't, you'll need to go in the database and change the `status` values of some users in the `office_users` table to `REQUESTED`
4. Once you see those users, click on one and you should see all the relevant information for that user (see screenshot)

## Screenshot
![Screenshot 2024-03-21 at 9 27 51 AM](https://github.com/transcom/mymove/assets/136510600/5a2ee858-ca6b-4466-9aa5-75583a761298)
